### PR TITLE
UITEN-93 Resolve ui-tenant quality gate failures

### PR DESCRIPTION
**Pull request by uladislausamets**
## Purpose

https://issues.folio.org/browse/UITEN-93
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by axelhunn** [comment]
@ViktorSoroka07 what do you think if we import jest config and mocks from `stripes-acq-components` here?
Instead of keeping the fork of them

File: src/settings/LocationLocations/locationDetail.test.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      stripes={stripes}
+    />
+  </MemoryRouter>
+));
+
+describe('LocationDetail', () => {
+  it('should render all fields in location details', () => {
+    renderLocationDetail();
+
+    expect(screen.queryByText('ui-tenant-settings.settings.location.institutions.institution')).toBeVisible();
```
Comment (axelhunn) [inline src/settings/LocationLocations/locationDetail.test.js:69]:
```suggestion
    expect(screen.getByText(/institutions.institution/)).toBeVisible();
```

Comment (axelhunn) [reply src/settings/LocationLocations/locationDetail.test.js:69]: `queryBy` is strongly recommended to use only for checking for non-existence.
See https://kentcdodds.com/blog/common-mistakes-with-react-testing-library/#using-query-variants-for-anything-except-checking-for-non-existence

And we can shorten the search criteria by using regex - this way we can drop identical part

File: src/settings/LocationLocations/locationDetail.test.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    expect(screen.queryByText('ui-tenant-settings.settings.location.locations.locationDetails')).toBeVisible();
+  });
+
+  it('should open delete confirmation window when delete button in action menu clicked', () => {
+    renderLocationDetail();
+    const deleteButton = screen.queryByText('stripes-core.button.delete');
+
+    user.click(deleteButton);
+    expect(screen.queryByText('ui-tenant-settings.settings.location.locations.deleteLocation')).toBeInTheDocument();
+  });
```
Comment (axelhunn) [inline src/settings/LocationLocations/locationDetail.test.js:87]:
```suggestion
  it('should open delete confirmation modal when delete button in action menu is clicked', () => {
    renderLocationDetail();
    const actions = screen.getByRole('button', { name: /actions/i });

    user.click(actions);
    const delete = screen.getByRole('button', { name: /delete/i });
    expect(delete).toBeVisible();

    user.click(delete);
    const dialog = screen.getByRole('dialog');
    expect(dialog).toBeVisible();
    expect(within(dialog).getByRole('heading', { name: /deleteLocation/ })).toBeVisible();
  });
```
This way we can use recommended "user interaction" and "find `byRole` first" approaches.

`within` is imported from `@testing-library/react`

I would also add `onRemove` call checking here.

File: jest.config.js [hunk 3]
```diff
@@ -12,13 +12,17 @@ module.exports = {
   coverageDirectory: './artifacts/coverage-jest/',
   coverageReporters: ['lcov'],
   reporters: ['jest-junit', 'default'],
-  transform: { '^.+\\.(js|jsx)$': path.join(__dirname, './test/jest/jest-transformer.js') },
-  transformIgnorePatterns: [`/node_modules/(?!${esModules}|ky)`],
+  transform: {
+    '^.+\\.(js|jsx|ts|tsx)$': path.join(__dirname, './test/jest/jest-transformer.js'),
```
Comment (axelhunn) [inline jest.config.js:16]: For TS we need to also add `@babel/preset-typescript` preset to jest's `babel.config.js`

File: src/settings/LocationLocations/locationDetail.test.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    expect(screen.queryByText('ui-tenant-settings.settings.location.locations.description')).toBeVisible();
+    expect(screen.queryByText('ui-tenant-settings.settings.location.locations.locationDetails')).toBeVisible();
+  });
+
+  it('should open delete confirmation modal when delete button in action menu is clicked', () => {
+    renderLocationDetail();
+    const actions = screen.getByRole('button', { name: /actions/i });
+
+    user.click(actions);
+    const delete = screen.getByRole('button', { name: /delete/i });
```
Comment (axelhunn) [inline src/settings/LocationLocations/locationDetail.test.js:86]: yes, I'm sorry, `delete` is a reserved keyword

**Comment by uladislausamets** [comment]
@ViktorSoroka07 could you review this one, please

**Review by ViktorSoroka07 — APPROVED** [review]

**Review by axelhunn — APPROVED** [review]

**Pull request merged**